src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/Models/ApprenticeshipIncentive.cs

```diff
@@ -41,6 +41,8 @@ public partial class ApprenticeshipIncentive
         [Column(TypeName = "nvarchar(50)")]
         public IncentiveStatus Status { get; set; }
 
+        public int? MinimumAgreementVersion { get; set; }
```

Comment:
Can you create a value object to hold this business logic.  e.g. an AgreementVersion value type with a static Calculate method.  Then in the incentive constructor  MinimumAgreementVersion = AgreementVersion.Calculate(DateTime startDate)

where Calculate returns a new AgreementVersion(int version) value type object?

That way the agreement logic is all contained in the value type and can be tested separately.

Also rather than expose the version expose the object e.g.

public AgreementVersion AgreementVersion { get; set; }

in fact does this need to be exposed yet?  If it's not used or required by anything else then should be kept private to the domain object.